project/dexter-eclipse/src/java/com/samsung/sec/dexter/eclipse/action/CreateSnapshotActionDelegate.java

```diff
@@ -178,11 +179,13 @@ private void addResourceAsTargetFile(final IResource resource){
 		try {
         	if(resource instanceof IFile){
         		final IFile targetFile = (IFile) resource;
-				if(targetFile.getName().endsWith(".java")){
+        
+        		if(EclipseUtil.isValidJavaResource(resource) || EclipseUtil.isValidCAndCppResource(resource)){
```

Comment:
EclipseUtil.isValidJavaResource(). This method checks whether resource is an instance of IFile.
So there is no need to check again at line 180.

Also you can remove nested if blocks like this:
if(EclipseUtil.isValid...() == false && EclipseUtil.isValid...() == false)
    return;

if(!targetFiles.contains(targetFile)) 
   targetFiles.add(targetFile);